components/calendar/src/indian.rs

```diff
@@ -0,0 +1,185 @@
+            31
+        } else if (7..=12).contains(&month) {
+            30
+        } else if month == 1 {
+            if Self::is_leap_year(year) {
+                31
+            } else {
+                30
+            }
+        } else {
+            0
+        }
+    }
+
+    fn months_for_every_year() -> u8 {
+        12
+    }
+
+    fn is_leap_year(year: i32) -> bool {
+        Iso::is_leap_year(year + 78)
+    }
+}
+
+impl Calendar for Indian {
+    type DateInner = IndianDateInner;
+    fn date_from_iso(&self, iso: Date<Indian>) -> DateInner {
+        let fixed_iso = Iso::fixed_from_iso(*iso.inner());
+        Self::indian_from_fixed(fixed_iso)
+    }
+
+    fn date_to_iso(&self, date: &Self::DateInner) -> Date<Indian> {
+        let fixed_indian = Indian::fixed_from_indian(date.0);
+        Iso::iso_from_fixed(fixed_indian)
+    }
+
+    fn months_in_year(&self, date: &Self::DateInner) -> u8 {
+        date.0.months_in_year()
+    }
+
+    fn days_in_year(&self, date: &Self::DateInner) -> u32 {
+        date.0.days_in_year()
+    }
+
+    fn days_in_month(&self, date: &Self::DateInner) -> u8 {
+        date.0.days_in_month()
+    }
+
+    fn day_of_week(&self, date: &Self::DateInner) -> types::IsoWeekday {
+        Iso.day_of_week(Indian.date_to_iso(date).inner())
+    }
+
+    fn offset_date(&self, date: &mut Self::DateInner, offset: DateDuration<Self>) {
+        date.0.offset_date(offset);
+    }
+
+    #[allow(clippy::field_reassign_with_default)]
+    fn until(
+        &self,
+        date1: &Self::DateInner,
+        date2: &Self::DateInner,
+        _largest_unit: DateDurationUnit,
+        _smallest_unit: DateDurationUnit,
+    ) -> DateDuration<Self> {
+        date1.0.until(date2.0, _largest_unit, _smallest_unit)
+    }
+
+    fn year(&self, date: &Self::DateInner) -> types::Year {
+        crate::gregorian::year_as_gregorian(date.0.year)
+    }
+
+    fn month(&self, date: &Self::DateInner) -> types::Month {
+        types::Month {
+            number: date.0.month.into(),
+            code: types::MonthCode(tinystr!(8, "TODO")),
+        }
+    }
+
+    fn day_of_month(&self, date: &Self::DateInner) -> types::DayOfMonth {
+        date.0.day_of_month()
+    }
+
+    fn day_of_year_info(&self, date: &Self::DateInner) -> types::DayOfYearInfo {
+        let prev_year = IsoYear(date.0.year - 1);
+        let next_year = IsoYear(date.0.year + 1);
+        types::DayOfYearInfo {
+            day_of_year: date.0.day_of_year(),
+            days_in_year: date.0.days_in_year(),
+            prev_year: prev_year.into(),
+            days_in_prev_year: Indian::days_in_year_direct(prev_year.0),
+            next_year: next_year.into(),
+        }
+    }
+
+    fn debug_name() -> &'static str {
+        "Indian"
+    }
+}
+
+impl Indian {
+    /// Construct a new Indian Calendar
+    pub fn new() -> Self {
+        Self
+    }
+
+    fn fixed_from_indian(date: ArithmeticDate<Indian>) -> i32 {
+        // TODO
```

Comment:
@Manishearth any good reference for this? Neither the lisp code nor the calendarical calculations has this

Reply:
Oh, it's ISO year - 78.

The Indian National calendar is the Gregorian calendar with a different year offset and different month numberings